### DISCUSSION TITLE
Progress on Replica Exchange

### DISCUSSION THREAD

**Discussion by ConnorLuellen** [blackjax-devs/blackjax#534] [General]
First, I'd like to thank everyone who's working on this project as it has saved me an astounding amount of time. 

Second, I'd like to ask about a feature which I've seen discussed as a possible implementation for this project: Parallel Tempering (replica exchange). I see that y'all are currently working on implementing this feature, but I'm currently working on a project which requires replica exchange, so I wanted to see if y'all have a prototype version of this code which I could use as a starting point before I set out on creating the function myself. 

Thanks, so much

**Comment by junpenglao** [comment]
Yeah it would be great to have replica exchange - it is awesome for sampling multi-modal density.

**Comment by albcab** [comment]
I don't think anybody has written any code for parallel tempering yet. You could open an issue with your proposal and we can discuss ideas on the implementation. Since you are doing a sampling algorithm, [here](https://github.com/blackjax-devs/blackjax/blob/4b9d9804b8143d7325c72950d94d844430d3f840/docs/developer/sampling_algorithm.py) is a skeleton you can use to build your algorithm. Also a good idea to mimic what is done in a similar algorithm like [tempered SMC](https://github.com/blackjax-devs/blackjax/blob/main/blackjax/smc/tempered.py).
